src/Cache/SimpleCacheClient.php

```diff
@@ -20,9 +22,21 @@ function __construct(string $authToken, int $defaultTtlSeconds, int $dataClientO
         );
     }
 
+    private function throwBadAuthToken() {
+        throw new InvalidArgumentError('Invalid Auth token.');
+    }
+
     private function parseAuthToken(string $authToken) : array {
-        list($header, $payload, $signature) = explode (".", $authToken);
-        return json_decode(base64_decode($payload), true);
+        $exploded = explode (".", $authToken);
+        if (count($exploded) != 3) {
```

Comment:
I think this is ok for now if we want. Feels like getting pretty close to where we should just use something off the shelf like https://github.com/firebase/php-jwt instead. @pgautier404 could you just make a follow ticket to track this so we dont loose sight of it can come back and revisit.

Reply:
Agreed, I'll make a ticket. This approach was just for maximum speed of delivery, as there were a number of PHP JWT libs to sift through.